Sources/Swift/Metrics/SetMetric.swift

```diff
@@ -2,18 +2,21 @@ import Foundation
 
 class SetMetric: Metric {
 
-    private var set: Set<Int32>
+    private var set: Set<UInt>
     var weight: UInt {
         return UInt(set.count)
     }
 
-    init(first: Int32, key: String, unit: MeasurementUnit, tags: [String: String]) {
+    init(first: UInt, key: String, unit: MeasurementUnit, tags: [String: String]) {
         set = [first]
         super.init(type: .set, key: key, unit: unit, tags: tags)
     }
 
+    // This doesn't work with the full range of UInt.
+    // We still need to fix this.
```

Comment:
l: If we really need the function signature to expect a Double, this is a limitation we will need to accept.

But now that I'm understanding this a little better, having `Metric.Add` always expect a double seem a little bit wrong.

Reply:
> But now that I'm understanding this a little better, having Metric.Add always expect a double seem a little bit wrong.

Yes, it is. It can be a double or an int.

I already have a PR for that https://github.com/getsentry/sentry-cocoa/pull/3795.